internal/server/event_server.go

```diff
@@ -222,3 +230,22 @@ func eventKeyFunc(r *http.Request) (string, error) {
 	digest := sha256.Sum256([]byte(key))
 	return fmt.Sprintf("%x", digest), nil
 }
+
+// EventFromContext extracts an *eventv1.Event from the given context,
+// or nil if not present.
+func EventFromContext(ctx context.Context) (event *eventv1.Event) {
+	if val := ctx.Value(eventContextKey{}); val != nil {
+		event = val.(*eventv1.Event)
+	}
+	return
+}
+
+// EventLoggerFromContext extracts a logr.Logger from the given context
+// enhanced with an event's involved object's kind, name and namespace,
+// or an empty logger if not present.
+func EventLoggerFromContext(ctx context.Context) (logger logr.Logger) {
+	if val := ctx.Value(eventLoggerContextKey{}); val != nil {
+		logger = val.(logr.Logger)
+	}
+	return
```

Comment:
same here:

```suggestion
func EventLoggerFromContext(ctx context.Context) (logr.Logger) {
	if val := ctx.Value(eventLoggerContextKey{}); val != nil {
		return val.(logr.Logger)
	}
	return logr.Logger{}
```

OTOH I'm not sure if it even makes sense to return an empty logger here instead of nil. 🤔

Reply:
I don't think we should return nil as that could lead to panic (when consumers of the function forget to check for nil). 

Instead, we can return a default logger instead of `logr.Logger{}` as it is basically a no-op.

Reply:
Right now the code is doing exactly what @somtochiama said. The `(logger logr.Logger)` return type creates a local default `logr.Logger` in the variable named `logger`. That variable is only ever updated if `ctx.Value(key)` returns a non-nil `any`, which is how the built-in `context.Context.Value()` reports that a value for that key was present.

Reply:
The patch suggested by @makkes here accomplishes exactly the same but without the `logger` local var created by the `(logger logr.Logger)` return type, i.e. it's just stylistic

Reply:
> we can return a default logger instead of `logr.Logger{}`

@somtochiama do you mean like a specific default logger created in a global var or by calling some `github.com/go-logr/logr` function? Otherwise I can only think of `logr.Logger{}`